### PR TITLE
feat(ctb): `toNibbles` optimization.

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Revisits #4340 after _6 months_ of the change being differential tested against the current implementation and an added memory safety cheatcode.

**Description**

Saves roughly ~15.5% of the cost of an average `proveWithdrawalTransaction` call to the `OptimismPortal`.

**Tests**

* Memory safety test for the inline-asm version of the `toNibbles` function, ported from [substratum](https://github.com/clabby/substratum)

**Additional context**
This change has never failed a differential test after months of inclusion in every single CI run. In addition, we now have memory safety assertions by forge.

### CODE REVIEW

File: op-bindings/bindings/optimismportal.go
Comment (maurelian) [inline op-bindings/bindings/optimismportal.go:1]: Portal is gonna need a semver bump

Comment (clabby) [reply op-bindings/bindings/optimismportal.go:1]: Good callout, bumped.

**Comment by tynes** [comment]
> This change has never failed a differential test after months of inclusion in every single CI run

Looks good to me

**Review by tynes — APPROVED** [review]

**Comment by maurelian** [comment]
> This change has never failed a differential test after months of inclusion in every single CI run

Where is the diff testing setup for this? Was there any difference between the first and the 1000th time that test ran?

**Pull request merged**